server/image.go

```diff
@@ -1,23 +1,91 @@
 package server
 
 import (
+	"errors"
+	"os"
+
+	"github.com/containers/image/image"
+	"github.com/containers/image/transports"
 	pb "github.com/kubernetes/kubernetes/pkg/kubelet/api/v1alpha1/runtime"
 	"golang.org/x/net/context"
 )
 
 // ListImages lists existing images.
 func (s *Server) ListImages(ctx context.Context, req *pb.ListImagesRequest) (*pb.ListImagesResponse, error) {
+	// TODO
+	// containers/storage will take care of this by looking inside /var/lib/ocid/images
+	// and listing images.
 	return nil, nil
 }
 
 // ImageStatus returns the status of the image.
 func (s *Server) ImageStatus(ctx context.Context, req *pb.ImageStatusRequest) (*pb.ImageStatusResponse, error) {
+	// TODO
+	// containers/storage will take care of this by looking inside /var/lib/ocid/images
+	// and getting the image status
 	return nil, nil
 }
 
 // PullImage pulls a image with authentication config.
 func (s *Server) PullImage(ctx context.Context, req *pb.PullImageRequest) (*pb.PullImageResponse, error) {
-	return nil, nil
+	img := req.GetImage().GetImage()
+	if img == "" {
+		return nil, errors.New("got empty imagespec name")
+	}
+
+	// TODO(runcom): deal with AuthConfig in req.GetAuth()
+
+	// TODO(mrunalp,runcom): why do we need the SandboxConfig here?
+	// how do we pull in a specified sandbox?
+	tr, err := transports.ParseImageName(img)
+	if err != nil {
+		return nil, err
+	}
+	src, err := tr.NewImageSource("", false)
+	if err != nil {
+		return nil, err
+	}
+	i := image.FromSource(src, nil)
+	blobs, err := i.BlobDigests()
+	if err != nil {
+		return nil, err
+	}
+	// TODO: make sure this dir exists?
+	if err := os.Mkdir("/var/lib/ocid/images"+tr.StringWithinTransport(), 0755); err != nil {
+		return nil, err
+	}
+	dirTr, err := transports.ParseImageName("dir:/var/lib/ocid/images/" + tr.StringWithinTransport())
```

Comment:
This could just be `directory.NewReference(path)` without needing the `dir:` prefix. Not sure that it is cleaner, perhaps the consistency of using `transports.ParseImageName` for both the source and destination is better.